include/swift/AST/NameLookup.h

```diff
@@ -73,13 +73,15 @@ namespace swift {
     /// When finding \c bar() from the function body of \c foo(), \c BaseDC is
     /// the method \c foo().
     ///
-    /// \c BaseDC will be the method if \c self is needed for the lookup,
-    /// and will be the type if not.
-    /// In other words: If \c baseDC is a method, it means you found an instance
-    /// member and you should add an implicit 'self.' (Each method has its own
-    /// implicit self decl.) There's one other kind of non-method context that
-    /// has a 'self.' -- a lazy property initializer, which unlike a non-lazy
-    /// property can reference \c self) Hence: \code
+    /// \c BaseDC will be the type if \c self is not needed for the lookup. If
+    /// \c self is needed, \c baseDC will be either the method or a closure
+    /// which explicitly captured \c self.
+    /// In other words: If \c baseDC is a method or a closure, it means you
+    /// found an instance member and you should add an implicit 'self.' (Each
+    /// method has its own implicit self decl.) There's one other kind of
+    /// non-method, non-closure context that has a 'self.' -- a lazy property
+    /// initializer, which unlike a non-lazy property can reference \c self)
```

Comment:
Not your fault, but there should probably be a period, not a paren, at the end of this sentence.